source/includes/security/authentication.kt

```diff
@@ -0,0 +1,200 @@
+import com.mongodb.*
+import com.mongodb.kotlin.client.MongoClient
+import org.bson.BsonInt64
+import org.bson.Document
+
+// SCRAM Authentication
+// start-default-cred-string
+val mongoClient =
+    MongoClient.create("mongodb://<db_username>:<db_password>@<hostname>:<port>/?authSource=<authenticationDb>")
+// end-default-cred-string
+
+// start-default-mongo-cred
+val credential = MongoCredential.createCredential(
+    "<db_username>", "<authenticationDb>", "<db_password>".toCharArray()
+)
+val settings = MongoClientSettings.builder()
+        .applyToClusterSettings { builder: ClusterSettings.Builder ->
+            builder.hosts(
+                listOf(ServerAddress("<hostname>", "<port>"))
+            )
+        }
+        .credential(credential)
+        .build()
+
+val mongoClient = MongoClient.create(settings)
+// end-default-mongo-cred
+
+// start-scramsha256-cred-string
+val mongoClient =  
+    MongoClient.create("mongodb://<db_username>:<db_password>@<hostname>:<port>/?authSource=admin&authMechanism=SCRAM-SHA-256")
+// end-scramsha256-cred-string
+
+// start-scramsha256-mongo-cred
+val credential = MongoCredential.createScramSha256Credential(
+    "<db_username>", "<authenticationDb>", "<db_password>".toCharArray()
+)
+val settings = MongoClientSettings.builder()
+        .applyToClusterSettings { builder: ClusterSettings.Builder ->
+            builder.hosts(
+                listOf(ServerAddress("<hostname>", "<port>"))
+            )
+        }
+        .credential(credential)
+        .build()
+
+val mongoClient = MongoClient.create(settings)
+// end-scramsha256-mongo-cred
+
+// start-scramsha1-cred-string
+val mongoClient =
+    MongoClient.create("mongodb://<db_username>:<db_password>@<hostname>:<port>/?authSource=admin&authMechanism=SCRAM-SHA-1")
+// end-scramsha1-cred-string
+
+// start-scramsha1-mongo-cred
+val credential = MongoCredential.createScramSha1Credential(
+    "<db_username>", "<authenticationDb>", "<db_password>".toCharArray()
+)
+val settings = MongoClientSettings.builder()
+        .applyToClusterSettings { builder: ClusterSettings.Builder ->
+            builder.hosts(
+                listOf(ServerAddress("<hostname>", "<port>"))
```

Comment:
Port is an integer and is specified as <port> (without quotes) in the [Java documentation](https://www.mongodb.com/docs/drivers/java/sync/current/fundamentals/auth/). To maintain consistency, I suggest we follow the same format:
```suggestion
                listOf(ServerAddress("<hostname>", <port>))
```